server/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -161,8 +168,16 @@ protected void shardOperationOnPrimary(
     protected void dispatchedShardOperationOnPrimary(
         BulkShardRequest request,
         IndexShard primary,
-        ActionListener<PrimaryResult<BulkShardRequest, BulkShardResponse>> listener
+        ActionListener<PrimaryResult<BulkShardRequest, BulkShardResponse>> outerListener
     ) {
+        var listener = ActionListener.releaseBefore(
+            indexingPressure.trackPrimaryOperationExpansion(
+                primaryOperationCount(request),
```

Comment:
Isn't think incrementing primary operations count a second time? I think we just want to do expansion memory here?

Reply:
We have to pass the number of operations to increment the number of rejected operations if the expansion is beyond the current limit. (there's an if to only increment the rejections in the method)